### PR TITLE
Refactor tests for improved clarity and feedback

### PR DESCRIPTION
**Pull request by StephenHulme**
Fixes encounted as part of #1178 

#### Changes proposed in this pull request

- rename factory `pacbio_run` to `pacbio_generic_run` to make it clear that it is not tailored to an implied system
  - update tests
- add additional feedback in validation messages: now indicate the incorrect input that caused the validation failure
  - update tests

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: app/validators/well_positions_validator.rb [hunk 1]
```diff
@@ -25,8 +25,13 @@ def valid_positions
   def validate(record)
     well_positions = filtered(record.wells).collect(&:position)
 
-    return if (well_positions - valid_positions).empty?
+    invalid_positions = (well_positions - valid_positions)
 
-    record.errors.add(:wells, "must be in positions #{valid_positions.join(',')}")
+    return if invalid_positions.empty?
+
+    invalid_wells = invalid_positions.join(',')
```
Comment (stevieing) [inline app/validators/well_positions_validator.rb:32]: Do we need to create new variables here? Could we just do the join in the add as the variables are not used anywhere else?

File: spec/exchanges/run_csv/pacbio_sample_sheet_spec.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
     subject(:csv_string) { csv.payload }
 
     let(:plate)       { build(:pacbio_plate, wells:, plate_number: 1) }
-    let(:run)         { create(:pacbio_run, smrt_link_version:, plates: [plate]) }
+    let(:run)         { create(:pacbio_generic_run, smrt_link_version:, plates: [plate]) }
```
Comment (stevieing) [inline spec/exchanges/run_csv/pacbio_sample_sheet_spec.rb:10]: What is the relevance of changing this to pacbio_generic_run?

**Review by stevieing — APPROVED** [review]
We will need to get this UATed. Although this is a minor thing it is thing we need to check it is still working as expected.

**Review by BenTopping — APPROVED** [review]

**Comment by stevieing** [comment]
> UAT looks fine. The message is displaying is displaying correctly and showing the current (incorrect) order.
> Screenshot 2024-04-18 at 14 27 30
> 
> The duplication must be a bug, as it affects all the validation errors and not just the one changed in this PR.
> 
> Inspecting the network traffic, it seems to be coming from the Traction-Service responses:
> 
> ```json
>         {
>             "title": "plate 1 wells must be in a valid order, currently B1,D1",
>             "detail": "plates - plate 1 wells must be in a valid order, currently B1,D1",
>             "code": "100",
>             "source": {
>                 "pointer": "/data/relationships/plates"
>             },
>             "status": "422"
>         }
> ```

This might have been introduced because of the work we have done to introduce aliquots for run creation. @BenTopping and @seenanair . It might be that we need to reformat the errors to remove the duplication?

**Comment by BenTopping** [comment]
Yes looks like a bug, I think we are posting the message to two stores and the alert component is reading from both of them. Ill take a look.

**Pull request merged**